### PR TITLE
Feat(dockerfile): Upgrade to bullseye

### LINKED ISSUE #165: Bump to latest patch of golang in dockerfile
Don't know why dependabot hasn't done this...

### PR DESCRIPTION
**Pull request by ctlong**
### What is this change about?

Upgrades the cf-d-c-t docker image to have Debian bullseye as the base of the image.

Included some refactors and updates that I think should be relatively minor:
- No longer include the bbl repo at `var/repos/bosh-bootloader`.
- Remove duplicate package installations.
- Remove packages that don't appear to be needed for any specific reason.
- Use ruby 2.7.7 instead of ruby 2.5.4.

### Please provide contextual information.

Closes #165.

### Please check all that apply for this PR:

- [ ] introduces a new task
- [ ] changes an existing task
- [x] changes the Dockerfile
- [x] introduces a breaking change (other users will need to make manual changes when this is released)

### Did you update the README as appropriate for this change?

- [ ] YES
- [x] N/A

### What is the level of urgency for publishing this change?

- [ ] **Urgent** - unblocks current or future work
- [x] **Slightly Less than Urgent**

### Tag your pair, your PM, and/or team!

None

### CODE REVIEW

**Review by jochenehret — DISMISSED** [review]
I've built the image locally and found no problems. I can't tell however if it works in our pipelines, we'll have to try...

**Comment by ctlong** [comment]
Going to wait until tomorrow or the next day to merge in order to give folks time to react to my message in Slack.

**Comment by ctlong** [comment]
Had to rebase to resolve merge conflicts

**Review by jochenehret — APPROVED** [review]

**Pull request merged**
